bitcoin/examples/sighash.rs

```diff
@@ -0,0 +1,199 @@
+use bitcoin::{consensus, ecdsa, sighash, PublicKey, Script, ScriptBuf, Transaction};
+use hex_lit::hex;
+
+//These are real blockchain transactions examples of computing sighash for:
+// - P2WPKH
+// - P2MS 2of3
+// - P2SH 2of2 multisig
+// - P2WSH 2of2 multisig
+
+//run with: cargo run --example sighash
+
+//TODO add P2TR examples, ideally for both key-path and script-path spending
+
+fn main() {
+    test_sighash_p2wpkh();
+    test_sighash_p2ms_multisig_2x3();
+    test_sighash_p2sh_multisig_2x2();
+    test_sighash_p2wsh_multisig_2x2();
+}
```

Comment:
These are not tests, they don't assert anything and don't fail if the "test" fails.